packages/react-native-editor/__device-tests__/helpers/utils.js

```diff
@@ -195,6 +188,9 @@ const stopDriver = async ( driver ) => {
  * On iOS: "clear" is not defaulted to true because calling element.clear when a text is present takes a very long time (approx. 23 seconds)
  */
 const typeString = async ( driver, element, str, clear ) => {
+	if ( isKeycode( str ) ) {
+		return await pressKeycode( driver, getKeycode( str ) );
+	}
```

Comment:
When typing special characters like `\n`, instead of typing them into the text input, we'll execute them by pressing their associated keycode. For example, typing `\n` will result in pressing the Enter key. On Android, this behavior was already in place ([reference](https://github.com/WordPress/gutenberg/blob/1be8fa3c5f63a0c9a096478805665724417512aa/packages/react-native-editor/__device-tests__/helpers/utils.js#L233-L240)), but not on iOS as typing the new line character is breaking the line. However, after the RN upgrade, I noticed that on iOS now typing on an element always places the caret at the start. This is unexpected compared to the past behavior, where invoking `typeString` always appended the string to wherever the caret was previously placed.

I haven't explored further what changed after the RN upgrade, but I found it more reliable to use keycodes for special characters instead of typing them.

**Example:**

| After | Before |
|--------|--------|
| <video src=https://github.com/WordPress/gutenberg/assets/14905380/eacf3282-c12a-4cd1-bb3f-4d77a78bf380> | <video src=https://github.com/WordPress/gutenberg/assets/14905380/e7f3ea57-d939-4f4b-b6c8-0163ad2e553f> |